sdk/servicebus/service-bus/src/core/messageReceiver.ts

```diff
@@ -867,89 +897,85 @@ export class MessageReceiver extends LinkEntity {
-        if (translatedError.retryable) {
-          shouldReopen = true;
-          log.error(
-            "[%s] close() method of Receiver '%s' with address '%s' was not called. There " +
-              "was an accompanying error and it is retryable. This is a candidate for re-establishing " +
-              "the receiver link.",
-            connectionId,
-            this.name,
-            this.address
-          );
-        } else {
-          log.error(
-            "[%s] close() method of Receiver '%s' with address '%s' was not called. There " +
-              "was an accompanying error and it is NOT retryable. Hence NOT re-establishing " +
-              "the receiver link.",
-            connectionId,
-            this.name,
-            this.address
-          );
-        }
-      } else if (!wasCloseInitiated) {
-        shouldReopen = true;
+      const translatedError = receiverError ? translate(receiverError) : receiverError;
+
+      // Track-1
+      //   - We should only attempt to reopen if either no error was present,
+      //     or the error is considered retryable.
+      // Track-2
+      //  Reopen
+      //   - If no error was present
+      //   - If the error is considered retryable
+      //   - Any errors that were triggered by a disconnect,
+      //     we don't care if the retryable flag is set to true on the error
+      //     because `@azure/core-amqp` is stricter in setting the retryable flag to true.
+      //   - More details here - https://github.com/Azure/azure-sdk-for-js/pull/8580#discussion_r417087030
+      const shouldReopen =
+        !translatedError ||
+        (isMessagingError(translatedError) ? translatedError.retryable : false) ||
+        causedByDisconnect;
+
+      // Non-retryable errors that aren't caused by disconnect
+      // will have already been forwarded to the user's error handler.
+      // Swallow the error and return quickly.
+      if (!shouldReopen && !causedByDisconnect) {
         log.error(
-          "[%s] close() method of Receiver '%s' with address '%s' was not called. " +
-            "There was no accompanying error as well. This is a candidate for re-establishing " +
-            "the receiver link.",
+          "[%s] Encountered a non retryable error on the receiver. Cannot recover receiver '%s' with address '%s' encountered error: %O",
           connectionId,
           this.name,
-          this.address
+          this.address,
+          translatedError
         );
-      } else {
-        const state: any = {
-          wasCloseInitiated: wasCloseInitiated,
-          receiverError: receiverError,
-          _receiver: this._receiver
-        };
+        return;
+      }
+
+      // Non-retryable errors that are caused by disconnect
+      // haven't had a chance to show up in the user's error handler.
+      // Rethrow the error so the surrounding try/catch forwards it appropriately.
+      if (!shouldReopen && causedByDisconnect) {
```

Comment:
This if condition will never be true right?

Reply:
Yes, both theoretically and logically.
But I thought to leave it there for a future where we plan on classifying the errors properly.
- Should I remove this if-section completely?
- Should I add a comment saying why this if-section would never get executed?
- Should I add a comment with an explanation as well as comment out this if-section?

Reply:
If we go with the change in https://github.com/Azure/azure-sdk-for-js/pull/8562/files#r418652131, then this comment thread can be ignored

Reply:
Done with that change. Resolving this.

Closing this PR in favor of the unified PR #8580.
Please continue over there.